src/playlist-loader/playlist-loader.js

```diff
@@ -0,0 +1,337 @@
+   */
+  request() {
+    return this.request_;
+  }
+
+  /**
+   * A getter for the uri string.
+   *
+   * @return {string}
+   *         The uri.
+   */
+  uri() {
+    return this.uri_;
+  }
+
+  /**
+   * A getter for the manifest object.
+   *
+   * @return {Object|null}
+   *         The manifest or null.
+   */
+  manifest() {
+    return this.manifest_;
+  }
+
+  /**
+   * Determine if the loader is started or not.
+   *
+   * @return {boolean}
+   *         True if stared, false otherwise.
+   */
+  started() {
+    return this.started_;
+  }
+
+  /**
+   * The last time a request happened.
+   *
+   * @return {number|null}
+   *         The last request time or null.
+   */
+  lastRequestTime() {
+    return this.lastRequestTime_;
+  }
+
+  /**
+   * A function that is called to when the manifest should be
+   * re-requested and parsed.
+   *
+   * @listens {PlaylistLoader#updated}
+   * @private
+   */
+  refreshManifest_() {
+    this.makeRequest_({uri: this.uri()}, (request, wasRedirected) => {
+      if (wasRedirected) {
+        this.uri_ = request.responseURL;
+      }
+
+      if (request.responseHeaders && request.responseHeaders.date) {
+        this.lastRequestTime_ = Date.parse(request.responseHeaders.date);
+      } else {
+        this.lastRequestTime_ = Date.now();
+      }
+
+      this.parseManifest_(request.responseText, (parsedManifest, updated) => {
+        if (updated) {
+          this.manifestString_ = request.responseText;
+          this.manifest_ = parsedManifest;
+          this.trigger('updated');
+        }
+      });
+    });
+  }
+
+  /**
+   * A function that is called to when the manifest should be
+   * parsed and merged.
+   *
+   * @param {string} manifestText
+   *        The text of the manifest directly from a request.
+   *
+   * @param {Function} callback
+   *        The callback that takes two arguments. The parsed
+   *        and merged manifest, and weather or not that manifest
+   *        was updated.
+   *
+   * @private
+   */
+  parseManifest_(manifestText, callback) {}
+
+  /**
+   * A function that is called when a playlist loader needs to
+   * make a request of any kind. Uses `withCredentials` from the
+   * constructor, but can be overriden if needed.
+   *
+   * @param {Object} options
+   *        Options for the request.
+   *
+   * @param {string} options.uri
+   *        The uri to request.
+   *
+   * @param {boolean} [options.handleErrors=true]
+   *        If errors should trigger on the playlist loader. If
+   *        This is false, errors will be passed along.
+   *
+   * @param {boolean} [options.withCredentials=false]
+   *        If this request should be sent withCredentials. Defaults
+   *        to the value passed in the constructor or false.
+   *
+   * @param {Function} callback
+   *        The callback that takes three arguments. 1 the request,
+   *        2 if we were redirected, and 3 error
+   *
+   * @private
+   */
+  makeRequest_(options, callback) {
```

Comment:
A generic request function with shared error handling. This allows stop to cancel any pending request.